py-polars/polars/plugins.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from polars._utils.parse_expr_input import parse_as_list_of_expressions
+from polars._utils.udfs import get_dynamic_lib_location
+from polars._utils.unstable import unstable
+from polars._utils.wrap import wrap_expr
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from polars import Expr
+    from polars.type_aliases import IntoExpr
+
+__all__ = ["register_plugin"]
+
+
+@unstable()
+def register_plugin(
+    *inputs: IntoExpr | list[IntoExpr],
```

Comment:
The `*args` syntax doesn't make much sense to me here. I would expect the `symbol` first and `args`/`kwargs` next to each other.

Let's follow the existing API with `plugin_location` / `symbol` / `args` / `kwargs`?